src/H5Cimage.c

```diff
@@ -2546,6 +2590,50 @@ H5C__reconstruct_cache_contents(H5F_t *f, H5C_t *cache_ptr)
     } /* end if */
 
 done:
+    if (FAIL == ret_value && pf_entry_ptr) {
+#ifndef NDEBUG
+        /* If we failed during reconstruction, remove reconstructed entries */
+        H5C_recon_entry_t *recon_entry, *tmp;
+
+        HASH_ITER(hh, recon_table, recon_entry, tmp)
+        {
+            H5C_cache_entry_t *entry_ptr = recon_entry->entry_ptr;
+
+            /* Only touch entries from the image reconstruction */
+            if (entry_ptr->type->id == H5AC_PREFETCHED_ENTRY_ID) {
```

Comment:
What is the purpose of this check? Shouldn't only entries from the cache image be in the hash table? Also I'm not sure if the entries will have this type after they're  reconstructed

Reply:
Should this be an assertion instead?